### PR TITLE
PIX passes: Centralize handle-generation code and update for 6.6

### PR DESCRIPTION
**Pull request by jeffnn**
This is little more than a move of the "create-uav" code in lib/DxilPIXPasses/DxilShaderAccessTracking.cpp to PixPassHelpers.cpp, followed by a factoring-out of the parts that create a handle (either pre-SM6.6 fashion, or with the newer create-from-binding etc.).
All the other passes' near-identical code was then deleted and made to call the centralized function.

### CODE REVIEW

**Comment by jeffnn** [comment]
@pow2clk: I have committed the faux-pas that you asked me not to here, by moving code and then modifying it. However, it has moved across files, so I don't think I could have made the diffs any easier...

File: lib/DxilPIXPasses/PixPassHelpers.cpp [hunk 1]
```diff
@@ -8,15 +8,151 @@
...
+
+  OP *HlslOP = DM.GetOP();
+  LLVMContext &Ctx = DM.GetModule()->getContext();
+
+  DXIL::ResourceClass resourceClass = resource->GetClass();
+
+  unsigned int resourceMetaDataId =
+      GetNextRegisterIdForClass(DM, resourceClass);
+
+  // Create handle for the newly-added UAV
```
Comment (pow2clk) [inline lib/DxilPIXPasses/PixPassHelpers.cpp:65]: Minor: Not necessarily a UAV

File: lib/DxilPIXPasses/PixPassHelpers.cpp [hunk 2]
```diff
@@ -8,15 +8,151 @@
...
+        resource_helper::getAsConstant(RP, resPropertyTy, *DM.GetShaderModel());
+
+    return Builder.CreateCall(annotHandleFn,
+                              {annotHandleArg, handle, propertiesV});
+  } else {
+    Function *CreateHandleOpFunc =
+        HlslOP->GetOpFunc(DXIL::OpCode::CreateHandle, Type::getVoidTy(Ctx));
+    Constant *CreateHandleOpcodeArg =
+        HlslOP->GetU32Const((unsigned)DXIL::OpCode::CreateHandle);
+    Constant *UAVArg = HlslOP->GetI8Const(
```
Comment (pow2clk) [inline lib/DxilPIXPasses/PixPassHelpers.cpp:104]: As above. Might not be UAV

**Review by pow2clk — APPROVED** [review]
Slight comment/varname suggestion, but not crucial

File: lib/DxilPIXPasses/PixPassHelpers.cpp (hunk 1 above)
Comment (jeffnn) [reply lib/DxilPIXPasses/PixPassHelpers.cpp]: Good piont!

**Pull request merged**